src/warnings.js

```diff
@@ -64,7 +64,20 @@ function lint(doc, position, prev, getErrorWords) {
             // Scan text nodes for bad words
             let m;
             while ((m = badWordsRegex.exec(node.text))) {
-                record(pos + m.index, pos + m.index + m[0].length, m[0]);
+                const matchingErrorWords = words.filter((word) => word.value === m[0]);
```

Comment:
@donspaulding I dont remember discussing these changes, were they fixing another bug we didnt talk abuot?